simpleclient/src/main/java/io/prometheus/client/Collector.java

```diff
@@ -91,8 +97,16 @@ public boolean equals(Object obj) {
           return false;
         }
         Sample other = (Sample) obj;
+        boolean timestampEquals = true;
```

Comment:
This is the Java 7 java.utils.Objects.equals. This project maintains Java 6 compatibility but it might be tidier to paste in this method and use it compare all the member objects using this method. 

`static boolean equals(Object a, Object b) {
        return (a == b) || (a != null && a.equals(b));
    }`

Reply:
I don't totally understand what you say:
```
 boolean timestampEquals = true;
        if ((timestamp != null && other.timestamp == null) || (timestamp == null && other.timestamp != null)) {
        	timestampEquals = false;
        }
        if (timestamp != null && other.timestamp != null) {
        	timestampEquals = other.timestamp.equals(timestamp);
        }
```
The `equals()` used here is the same used in other parts of previously existing code, such as:
```
return other.name.equals(name) && other.labelNames.equals(labelNames)
          && other.labelValues.equals(labelValues) && other.value == value;
```

Reply:
My suggestion is to based on readability - but it's just a suggestion.